MRG, FIX: prettier "head" extrapolation

mmagnuski opened the pull request:
Fixes #8425

larsoner commented on mne/viz/topomap.py:
```diff
@@ -568,7 +568,10 @@ def _get_extra_points(pos, extrapolate, origin, radii):
         assert extrapolate == 'head'
         # return points on the head circle
-        angle = np.arcsin(distance / 2 / np.mean(radii))
+        angle = np.arcsin(distance / np.mean(radii))
+        n_pnts = np.round(2 * np.pi / angle)
+        angle = 2 * np.pi / n_pnts
+
         points_l = np.arange(0, 2 * np.pi, angle)
```
At this point I would just do `np.linspace(0, 2 * np.pi, n_pts, endpoint=False)` -- it ensures you get exactly `n_pts` which you are not guaranteed from the `np.arange` operating on `float` due to precision issues, and you don't need to recompute `angle`.

mmagnuski commented:
Update:
I've increased the distance between extrapolation points and the channels:

<img src=https://user-images.githubusercontent.com/8452354/101987271-5347b700-3c93-11eb-8570-80ee66d6ff24.PNG width="250pix">


and it looks good now. Below is the history:

### before ('box' interpolation)
![image](https://user-images.githubusercontent.com/8452354/97121321-b7cba880-171d-11eb-9030-45863668659e.png)

### master ('head' with not-so-good extrapolation points placement)
![image](https://user-images.githubusercontent.com/8452354/97121378-0711d900-171e-11eb-9e1c-d88ed2d6b53c.png)

### this PR ('head' with corrected extrapolation points placement)
![topo now v2 cut](https://user-images.githubusercontent.com/8452354/101986437-2349e500-3c8e-11eb-8ab7-e4f1244af181.PNG)

mmagnuski commented:
One test fails and I now remember why I increased number of extrapolation points. But I think this could be easily fixed by using `max(n_pnts, 8)` as the number of exrapolation points for example. I'll test that later.

larsoner commented:
I would go with 8 or 16 minimum

mmagnuski commented:
I just looked into the older PR where I changed the number of extrapolation points (#7258) and we had 12 extrapolation points with 5 channels back then. So I'll go with 12 as the min - no test change should be necessary then :)

larsoner commented on mne/viz/topomap.py:
```diff
@@ -568,9 +568,10 @@ def _get_extra_points(pos, extrapolate, origin, radii):
+        angle = np.arcsin(distance / np.mean(radii))
+        n_pnts = max(12, int(np.round(2 * np.pi / angle)))
+        points_l = np.linspace(0, 2 * np.pi, n_pnts, endpoint=False)
+        use_radii = radii * 1.1 + distance
```
@mmagnuski this `+ distance` was breaking the tests but I assume it was intentional, so I pushed a commit to adjust the test to be more lenient

mmagnuski replied:
:+1:

mmagnuski commented:
@larsoner Thanks! And sorry for being slow, I saw the error and planned to fix the test.

larsoner commented:
No problem, it was a quick fix and I want to make sure this gets into 0.22, which we want to release today or tomorrow

mmagnuski commented:
Ok, I fixed the merge conflict. The "Fix bug where events could overflow when writing to FIF" was present twice in the log so I removed the second instance.

larsoner commented:
Thanks @mmagnuski !

mmagnuski commented:
Thanks @larsoner! :rocket:
